lib/src/segy.c

```diff
@@ -1846,6 +1846,84 @@ int segy_traces( segy_datasource* ds,
     return SEGY_OK;
 }
 
+/* Allows only int2 and uint2. */
+static int field_data_to_scalar(
+    const segy_field_data* fd,
+    int* scalar
+) {
+    switch( fd->entry_type ) {
+        case SEGY_ENTRY_TYPE_INT2:
+            *scalar = fd->value.i16;
+            break;
+        case SEGY_ENTRY_TYPE_UINT2:
+            *scalar = fd->value.u16;
+            break;
+        default:
+            return SEGY_INVALID_FIELD_DATATYPE;
+    }
+    return SEGY_OK;
+}
+
+int segy_delay_recoding_time( segy_datasource* ds, float* delay ) {
+    char trace_header[SEGY_TRACE_HEADER_SIZE];
+
+    /* we don't need to figure out a trace size, since we're not advancing
+     * beyond the first header */
+    int err = segy_read_standard_traceheader( ds, 0, trace_header );
+    if( err != 0 ) {
+        return err;
+    }
+
+    const segy_entry_definition* standard_map =
+        ds->traceheader_mapping_standard.offset_to_entry_definition;
+    const uint8_t* standard_name_map =
+        ds->traceheader_mapping_standard.name_to_offset;
+
+    const int delay_offset = standard_name_map[SEGY_TR_DELAY_REC_TIME];
+    int delay_raw = 0;
+    segy_field_data delay_fd;
+    err = segy_get_tracefield(
+        trace_header, standard_map, delay_offset, &delay_fd
+    );
+    if( err != SEGY_OK ) return err;
+
+    // until someone complains, we restrict allowed types to these only
+    switch( delay_fd.entry_type ) {
+        case SEGY_ENTRY_TYPE_TIME2:
+        case SEGY_ENTRY_TYPE_INT2:
+            delay_raw = delay_fd.value.i16;
+            break;
+        case SEGY_ENTRY_TYPE_UINT2:
+            delay_raw = delay_fd.value.u16;
+            break;
+        default:
+            return SEGY_INVALID_FIELD_DATATYPE;
```

Comment:
Does it make sense to allow for `SEGY_ENTRY_TYPE_TIME2` such that we could call `field_data_to_scalar` here?

I am mainly wondering if we would have a good way to reuse `field_data_to_scalar` here instead of reimplementing (almost) the same method here.

Reply:
Yeah, what I wrote is confusing.

The implementation of `field_data_to_scalar` looks almost like this `switch` statement here. `field_data_to_scalar` is only missing the `SEGY_ENTRY_TYPE_TIME2 ` case. I was wondering if it would make sense to add the `SEGY_ENTRY_TYPE_TIME2` case to `field_data_to_scalar` such that we could call `field_data_to_scalar`.